src/Microsoft.AspNetCore.Http/PipeWriterAdapter.cs

```diff
@@ -0,0 +1,287 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Pipelines;
+using System.Runtime.CompilerServices;
+using System.Runtime.ExceptionServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Http
+{
+    /// <summary>
+    /// Implements PipeWriter using a base stream implementation. 
+    /// </summary>
+    public class PipeWriterAdapter : PipeWriter, IDisposable
```

Comment:
override WriteAsync and provide an optimized path that writes directly to the Stream if there's no buffered data.

Reply:
When you say "No buffered data", you mean no data in _completedSegments or CurrentSegment?